### ISSUE TITLE
Automate header comparison, automated adn-visits vs 'normal' selenium visits

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#875]

**Comment by cqx931** [comment]
Related tickets:
https://github.com/dhowe/AdNauseam/issues/863
https://github.com/dhowe/AdNauseam/issues/862

Creating automated test cases on selenium to compare headers from adn-visits and 'normal' selenium visits
Simple test case example:
1.Run selenium with adnauseam 
2.Go to [simple ad](http://lab-lamp.scm.cityu.edu.hk/adntest/simple.html) and click the a tag with selenium
3.Compare the headers from selenium click and adn-visits

Question: How to get headers from 'normal' selenium visit(through click)?
From my research on the web, selenium itself does not capture network traffic
So how are the headers captured in this case?Through chrome [webRequest api](https://developer.chrome.com/extensions/webRequest)?
If it is using webRequest api, is the comparison going to happen within AdNauseam?

**Comment by dhowe** [comment]
Check if you can access headers through the Selenium api...

**Comment by cqx931** [comment]
this one?
https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/http/index_exports_Request.html

**Comment by dhowe** [comment]
No. We are trying to compare our requests to normal browser requests, so using a request API doesn't help us. Instead use browsermob-proxy or similar:
https://github.com/lightbody/browsermob-proxy/#using-with-selenium

**Comment by cqx931** [comment]
When I test Browsermob-proxy with adnauseam, it doesn't capture the requests for ad visit. I have tried to open background.html, but there is no traffic from adnauseam ad visit.

**Comment by dhowe** [comment]
and idea why?

**Comment by cqx931** [comment]
Fixed the above problem, two simple issues were involved:
1. Enough waiting time for AdNauseam to visit the ad
2. The AdVault need to be cleared after each session because it is using an user profile
(Another solution would be to launch chrome web driver and forbid anything to be saved to the profile. But I didn't find the corresponding chrome option yet.)

Please check the first try for headerComparison:
https://github.com/dhowe/Miscellaneous/pull/10
To run the test, please replace the profilePath to your own adnauseam Testing Profile.
I checked in the whole project, but there might be some configuration issues that I missed. Please let me know whether you could run the test as expected.

**Comment by cqx931** [comment]
Any feedback on this? @dhowe

**Comment by dhowe** [comment]
@cqx931 can we retest this with 3.4 for both firefox and chrome...
